spring-session/src/main/java/org/springframework/session/data/redis/RedisOperationsSessionRepository.java

```diff
@@ -446,8 +459,8 @@ public void onMessage(Message message, byte[] pattern) {
 
 
 		if(channel.startsWith(getSessionCreatedChannelPrefix())) {
-			RedisSerializer<Object> serializer = new JdkSerializationRedisSerializer();
-			Map<Object,Object> loaded = (Map<Object, Object>) serializer.deserialize(message.getBody());
+			// TODO: is this thread safe?
+			Map<Object,Object> loaded = (Map<Object, Object>) defaultSerializer.deserialize(message.getBody());
```

Comment:
Actually I think this casting is wrong and was related to use of jdk serialization. Been playing with kryo and I'm now getting casting errors. Need to play with it a bit more.